spec/lib/query/proposal/search_spec.rb

```diff
@@ -45,22 +69,47 @@
       [:product_name_and_description, :justification, :additional_info].each do |attr_name|
         it "returns the Proposal when searching by the ##{attr_name}" do
           procurement = create(:gsa18f_procurement, attr_name => 'foo')
-          results = Query::Proposal::Search.new.execute('foo')
-          expect(results).to eq([procurement.proposal])
+          procurement.proposal.reindex
+          refresh_index
+          results = Query::Proposal::Search.new(current_user: procurement.requester).execute('foo')
+          expect(results.to_a).to eq([procurement.proposal])
         end
       end
     end
 
     it "returns the Proposals by rank" do
-      prop1 = create(:proposal, id: 12)
-      work_order = create(:ncr_work_order, project_title: "12 rolly chairs for 1600 Penn Ave")
-      prop2 = work_order.proposal
-      prop3 = create(:proposal, id: 1600)
-
-      searcher = Query::Proposal::Search.new
-      expect(searcher.execute('12')).to eq([prop1, prop2])
-      expect(searcher.execute('1600')).to eq([prop3, prop2])
-      expect(searcher.execute('12 rolly')).to eq([prop2])
+      user = create(:user, client_slug: "test")
+
+      proposal1 = create(:proposal, id: 199, requester: user)
+      test_client_request1 = create(:test_client_request, proposal: proposal1)
+      proposal1.reindex
+
+      test_client_request2 = create(:test_client_request, project_title: "199 rolly chairs for 1600 Penn Ave")
+      proposal2 = test_client_request2.proposal
+      test_client_request2.add_observer(user.email_address)
+      proposal2.reindex
+
+      proposal3 = create(:proposal, id: 1600, requester: user)
+      test_client_request3 = create(:test_client_request, proposal: proposal3)
+      proposal3.reindex
+
+      refresh_index
+
+      searcher = Query::Proposal::Search.new(current_user: user)
+      expect(searcher.execute('199').to_a).to eq([proposal1, proposal2])
+      expect(searcher.execute('1600').to_a).to eq([proposal3, proposal2])
```

Comment:
I think re-creating the necessary fixture data 3x would generate a lot of noise. I tried. The mechanics made the test harder to follow. If we were comfortable with code comments, this would be a natural place to use them. As it is, I added some words to the existing description to help clarify what is being tested.

That said, if you can suggest a way to re-write that test as you're describing, I would be interested to see it.